### DISCUSSION TITLE
Use LDAP as Datasource for Birt RPT-Designer All in One

### DISCUSSION THREAD

**Discussion by mabu999** [eclipse-birt/birt#1555] [Q&A] [answered]
Hello,

is there a way to use LDAP as Datasource for Birt RPT-Designer ?

Looking for LDAP-JDBC-Drivers I only found cdata.jdbc.ldap.LDAPDriver or very very old ones .

Running the cdata.jdbc.ldap.LDAPDriver standalone, I got a LDAP-Connection, within RPT-Designer 4.13, I got the Error Message "Failed to create a connection. Please verify the connection properties."

Regards,
Markus

**Comment by speckyspooky** [comment, marked as answer]
There is no default way for the LDAP-queries because the datasource implementation is based on the dao-concept (data access object).

This means the driver must fullfill the concept with the according API-methods to connect a data sorce, query the source and retrieve data from there. But this is a task of the driver and I never found an LDAP-driver for that.

My supplier implemented an own Java-extension/PlugIn for such topics and I use it on JavaScript level.
The additional problem is that the LDAP-tree structure is not a standard in his nodes so each LDAP use his own structure.
And so it wouldn't be so easy to create a jdbc-ldap-driver for that.

**Reply by mabu999** [reply]
Thank you
